libs/wingsdk/src/std/resource.ts

```diff
@@ -111,16 +102,50 @@ export interface IResource extends IConstruct, IHostedLiftable {
  */
 export abstract class Resource extends Construct implements IResource {
   /**
-   * Register that the resource type needs to be lifted to the host for the given
-   * operations. A type being lifted to a host means that that type's static members
-   * will be lifted to the host.
+   * A hook called by the Wing compiler once for each inflight host that needs to
+   * use this type inflight. The list of requested inflight methods
+   * needed by the inflight host are given by `ops`.
    *
+   * This method is commonly used for adding permissions, environment variables, or
+   * other capabilities to the inflight host.
+   */
+  public static onLiftType(host: IInflightHost, ops: string[]): void {
```

Comment:
I feel like we can just go ahead and call this `onLift` as well. Lifting a class (i.e. a function) in the end isn't really different than lifting anything else.

Reply:
The issue I had was JSII doesn't allow instance and static members to have the same name :S but otherwise I agree

Reply:
Maybe I can just remove this from the abstract class

Reply:
I think it's possible but it'll require adding some more codegen, in any case I'm thinking to defer it for now if that's ok